ethcore/sync/src/chain/propagator.rs

```diff
@@ -342,8 +343,8 @@ impl SyncPropagator {
 	}
 
 	/// Generic packet sender
-	pub fn send_packet(sync: &mut SyncIo, peer_id: PeerId, packet_id: PacketId, packet: Bytes) {
-		if let Err(e) = sync.send(peer_id, packet_id, packet) {
+	pub fn send_packet(sync: &mut SyncIo, protocol: ProtocolId, peer_id: PeerId, packet_id: PacketId, packet: Bytes) {
+		if let Err(e) = sync.send_protocol(protocol, peer_id, packet_id, packet) {
```

Comment:
Interesting find, afaict this change is actually breaking (i.e. `propagate_private_transaction` and `propagate_consensus_packet`) will now always use `WARP_PROTOCOL_ID` while before it was up to the network to decide on the protocol.
Anyone knows how it worked before and if we are not breaking anything here? CC @grbIzl

Reply:
Hmmm, I guess I made the assumption that:
```
pub const CONSENSUS_DATA_PACKET: u8 = 0x15;
pub const PRIVATE_TRANSACTION_PACKET: u8 = 0x16;`
```
somehow implied `WARP`, but sure, it's just an assumption which may not be true and break.

Reply:
@tomusdrw if i understand correctly, this logic relied upon the deleted if case in requester module. So with this change the sender specifies protocol explicitly but in principal nothing has changed.